src/app/parser/parser.ts

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable max-classes-per-file */
+
+export class ParsedFunction {
+  constructor(
+    public readonly name: string,
+    public readonly body: string,
+    public readonly args: string[],
+  ) {
+  }
+}
+
+class FunctionSignature {
+  constructor(
+    public readonly name: string,
+    public readonly args: string[],
+  ) {
+  }
+}
+
+/**
+ * Given a Python file, returns the indentation character sequence
+ * @param file file to get the indentation for
+ */
+const getIndentation = (file: string): string | null => {
```

Comment:
Here and below, is there a reason why we use `const funcName = () => {}` instead of `function funcName() {}`?

Reply:
this is ES6 format which is more recent, can use the old one

Reply:
Oh ok, let's leave it that way then!